clients/js/main.ts

```diff
@@ -267,20 +277,44 @@ yargs(hideBin(process.argv))
       ) {
         throw Error(`Unknown network: ${network}`);
       }
+      let chain = argv["chain"]
       let module = argv["module"] as "Core" | "NFTBridge" | "TokenBridge";
+      let addr = ""
       switch (module) {
         case "Core":
-          console.log(CONTRACTS[network][argv["chain"]]["core"]);
+          addr = CONTRACTS[network][chain]["core"];
           break;
         case "NFTBridge":
-          console.log(CONTRACTS[network][argv["chain"]]["nft_bridge"]);
+          addr = CONTRACTS[network][chain]["nft_bridge"];
           break;
         case "TokenBridge":
-          console.log(CONTRACTS[network][argv["chain"]]["token_bridge"]);
+          addr = CONTRACTS[network][chain]["token_bridge"];
           break;
         default:
           impossible(module);
       }
+      if (argv["emitter"]) {
+        if (chain === "solana" || chain === "pythnet") { // TODO: Create an isSolanaChain()
+          addr = await getEmitterAddressSolana(addr)
+        } else if (isTerraChain(chain)) {
+          addr = await getEmitterAddressTerra(addr)
+        } else if (chain === "algorand") {
+          if (network !== "MAINNET") {
+            throw Error(`unable to look up algorand emitter address for ${network}`);
+          }
+          addr = "25e716e0618d9f38b603a97cc42db659069c0f5185230e5e61679fa876191ec4"
```

Comment:
Just out of curiosity, why isn't prettier putting semicolons at the end of some lines?

Reply:
Dunno.